src/OpenLoco/Windows/CompanyWindow.cpp

```diff
@@ -2278,12 +2287,93 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
+        // for example 10.0% (performanceIndex=100) converts to rating "Engineer" (output=1)
+        // input:
+        // ax = performanceIndex
+        // output:
+        // ax
+        static int16_t sub_437D60(int16_t performanceIndex)
+        {
+            registers regs;
+            regs.ax = performanceIndex;
+            call(0x00437D60, regs);
+            return regs.ax;
+        }
+
         // this will prepare _commonFormatArgs array before drawing the StringIds::challenge_value
         // after that for example it will draw this string: Achieve a performance index of 10.0% ("Engineer")
         // 0x004384E9
-        static void sub_4384E9()
+        static FormatArguments sub_4384E9()
         {
-            call(0x004384E9);
+            FormatArguments args{};
+
+            // 004384ED
+            switch (objectiveType)
+            {
+                case 0: // 004384FB-0043850A
+                    args.push(StringIds::achieve_a_company_value_of);
+                    args.push(*objectiveCompanyValue);
+                    break;
+
+                case 1: // 0043850C-0043851B
+                    args.push(StringIds::achieve_a_monthly_profit_from_vehicles_of);
+                    args.push(*objectiveMonthlyVehicleProfit);
+                    break;
+
+                case 2: // 0043851D-00438543
+                {
+                    args.push(StringIds::achieve_a_performance_index_of);
+                    int16_t performanceIndex = objectivePerformanceIndex * 10;
+                    args.push(performanceIndex);
+                    string_id corporateRating = sub_437D60(performanceIndex) + StringIds::corporate_rating_platelayer;
+                    args.push(corporateRating);
+                    break;
+                }
+
+                case 3: //00438545-00438576
+                {
+                    args.push(StringIds::deliver);
+                    cargo_object* cargoObject = _50D15C;
+                    if (objectiveDeliveredCargoType != 0xFF)
+                    {
+                        cargoObject = _cargoObjects[objectiveDeliveredCargoType];
+                    }
+                    args.push(cargoObject->unit_name_plural);
+                    args.push(*objectiveDeliveredCargoAmount);
+                    break;
+                }
+            }
+
+            if ((objectiveFlags & Scenario::objective_flags::flag_0) != 0) // 0043857B
+            {
+                args.push(StringIds::and_be_the_top_performing_company);
+            }
+            if ((objectiveFlags & Scenario::objective_flags::be_top_company) != 0) // 0043858C
+            {
+                args.push(StringIds::and_be_one_of_the_top_3_performing_companies);
+            }
+            if ((objectiveFlags & Scenario::objective_flags::be_within_top_three_companies) != 0) // 0043859D
+            {
+                if (isTitleMode() || isEditorMode())
```

Comment:
Please double check if the expression is correct with this assembly line:
text:004385A6                 test    screen_flags, 3
.text:004385AF                 jnz     short loc_4385C5

Reply:
yes that is correct